### ISSUE TITLE
Cannot read property 'apiIsDown' of undefined

### ISSUE THREAD

**Issue by 1522402210** [wazuh/wazuh#354]
3.x

netstat -ltnp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 127.0.0.1:9200          0.0.0.0:*               LISTEN      4661/java           
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      2916/nginx: master  
tcp        0      0 0.0.0.0:8338            0.0.0.0:*               LISTEN      5126/python         
tcp        0      0 127.0.0.1:9300          0.0.0.0:*               LISTEN      4661/java           
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      2320/sshd           
tcp        0      0 0.0.0.0:5601            0.0.0.0:*               LISTEN      5188/node           
tcp        0      0 0.0.0.0:10050           0.0.0.0:*               LISTEN      1376/zabbix_agentd  
tcp6       0      0 :::55000                :::*                    LISTEN      31665/node          
tcp6       0      0 :::10050                :::*                    LISTEN      1376/zabbix_agentd 

I installed wazuh on centos7
everything is normal

But when visiting wazuh of kibana, hint:

![image](https://user-images.githubusercontent.com/25600994/34876818-41ff4852-f7de-11e7-988b-ffb4742f3758.png)

Cannot read property 'apiIsDown' of undefined

Where did you make a mistake? Thank you

**Comment by 1522402210** [comment]
service wazuh-api status
Redirecting to /bin/systemctl status  wazuh-api.service
● wazuh-api.service - Wazuh API daemon
   Loaded: loaded (/etc/systemd/system/wazuh-api.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2018-01-12 20:08:51 CST; 1h 23min ago
     Docs: http://wazuh-documentation.readthedocs.org/en/latest/ossec_api.html
 Main PID: 31665 (node)
   CGroup: /system.slice/wazuh-api.service
           └─31665 /bin/node /var/ossec/api/app.js

**Comment by 1522402210** [comment]
systemctl status wazuh-api.service
● wazuh-api.service - Wazuh API daemon
   Loaded: loaded (/etc/systemd/system/wazuh-api.service; enabled; vendor preset: disabled)
   Active: active (running) since Fri 2018-01-12 21:42:22 CST; 26s ago
     Docs: http://wazuh-documentation.readthedocs.org/en/latest/ossec_api.html
 Main PID: 15130 (node)
   CGroup: /system.slice/wazuh-api.service
           └─15130 /bin/node /var/ossec/api/app.js

Jan 12 21:42:22 waiplay-re.verse.proxy systemd[1]: wazuh-api.service failed.
Jan 12 21:42:22 waiplay-re.verse.proxy systemd[1]: Started Wazuh API daemon.
Jan 12 21:42:22 waiplay-re.verse.proxy systemd[1]: Starting Wazuh API daemon...

**Comment by 1522402210** [comment]
SavedObjectNotFound {message: "Could not locate that index-pattern (id: wazuh-ale…ent/kibana/index?id=wazuh-alerts-3.x-%252A&name=)", savedObjectType: "index-pattern", savedObjectId: "wazuh-alerts-3.x-*", stack: "Error: Could not locate that index-pattern (id: wa….75.23/bundles/commons.bundle.js?v=16350:35:39923"} "Possibly unhandled rejection: {"message":"Could not locate that index-pattern (id: wazuh-alerts-3.x-*), [click here to re-create it](#/management/kibana/index?id=wazuh-alerts-3.x-%252A&name=)","savedObjectType":"index-pattern","savedObjectId":"wazuh-alerts-3.x-*"}"
(anonymous) @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
processChecks @ commons.bundle.js?v=16350:35
$digest @ commons.bundle.js?v=16350:35
$apply @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
completeOutstandingRequest @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
setTimeout (async)
Browser.self.defer @ commons.bundle.js?v=16350:35
scheduleApplyAsync @ commons.bundle.js?v=16350:35
$applyAsync @ commons.bundle.js?v=16350:35
done @ commons.bundle.js?v=16350:35
completeRequest @ commons.bundle.js?v=16350:35
xhr.onload @ commons.bundle.js?v=16350:35
XMLHttpRequest.send (async)
(anonymous) @ commons.bundle.js?v=16350:35
sendReq @ commons.bundle.js?v=16350:35
serverRequest @ commons.bundle.js?v=16350:35
processQueue @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
$digest @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
completeOutstandingRequest @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
setTimeout (async)
Browser.self.defer @ commons.bundle.js?v=16350:35
$evalAsync @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
scheduleProcessQueue @ commons.bundle.js?v=16350:35
then @ commons.bundle.js?v=16350:35
chainInterceptors @ commons.bundle.js?v=16350:35
$http @ commons.bundle.js?v=16350:35
value @ wazuh.bundle.js?v=16350:37
value @ wazuh.bundle.js?v=16350:37
SavedObjectsClient.processBatchQueue._lodash2.default.throttle.leading @ wazuh.bundle.js?v=16350:37
complete @ commons.bundle.js?v=16350:7
maxDelayed @ commons.bundle.js?v=16350:7
setTimeout (async)
debounced @ commons.bundle.js?v=16350:7
(anonymous) @ wazuh.bundle.js?v=16350:37
$Q @ commons.bundle.js?v=16350:35
value @ wazuh.bundle.js?v=16350:37
value @ wazuh.bundle.js?v=16350:61
IndexPatternsProvider.self.make @ wazuh.bundle.js?v=16350:61
IndexPatternsProvider.self.get @ wazuh.bundle.js?v=16350:61
(anonymous) @ wazuh.bundle.js?v=16350:103
processQueue @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
$digest @ commons.bundle.js?v=16350:35
$apply @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
completeOutstandingRequest @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
setTimeout (async)
Browser.self.defer @ commons.bundle.js?v=16350:35
scheduleApplyAsync @ commons.bundle.js?v=16350:35
$applyAsync @ commons.bundle.js?v=16350:35
done @ commons.bundle.js?v=16350:35
completeRequest @ commons.bundle.js?v=16350:35
xhr.onload @ commons.bundle.js?v=16350:35
XMLHttpRequest.send (async)
(anonymous) @ commons.bundle.js?v=16350:35
sendReq @ commons.bundle.js?v=16350:35
serverRequest @ commons.bundle.js?v=16350:35
processQueue @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
$digest @ commons.bundle.js?v=16350:35
$apply @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
completeOutstandingRequest @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
setTimeout (async)
Browser.self.defer @ commons.bundle.js?v=16350:35
scheduleApplyAsync @ commons.bundle.js?v=16350:35
$applyAsync @ commons.bundle.js?v=16350:35
done @ commons.bundle.js?v=16350:35
completeRequest @ commons.bundle.js?v=16350:35
xhr.onload @ commons.bundle.js?v=16350:35
XMLHttpRequest.send (async)
(anonymous) @ commons.bundle.js?v=16350:35
sendReq @ commons.bundle.js?v=16350:35
serverRequest @ commons.bundle.js?v=16350:35
processQueue @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
$digest @ commons.bundle.js?v=16350:35
$apply @ commons.bundle.js?v=16350:35
(anonymous) @ commons.bundle.js?v=16350:35
invoke @ commons.bundle.js?v=16350:35
doBootstrap @ commons.bundle.js?v=16350:35
bootstrap @ commons.bundle.js?v=16350:35
chrome.bootstrap @ commons.bundle.js?v=16350:7
(anonymous) @ wazuh.bundle.js?v=16350:103
__webpack_require__ @ commons.bundle.js?v=16350:1
window.webpackJsonp @ commons.bundle.js?v=16350:1
(anonymous) @ wazuh.bundle.js?v=16350:1

**Comment by havidarou** [comment]
Hi @1522402210 ,

are you using some kind of tunnel or secure connection for the API?

Somehow the Wazuh app is not able to reach the API.

Regards.

**Comment by skytremor** [comment]
Same problem here. Not utilizing tunnel or secure connection for the API.

**Comment by jesusgn90** [comment]
Hi @1522402210 and @skytremor first place is to apologize about that error, the newer Wazuh Apps have being patched from it. It's not a mistake, it's an Elasticsearch related error that is falling on another function and showing an incorrect message, we have fixed it with the new Wazuh 3.2 release. 

Any case you are facing that trouble due to a wrong Elasticsearch environment. The message `Could not locate that index-pattern (id: wazuh-alerts-3.x-` is the key here. Wazuh stores the alerts in indices that starts with a prefix like `wazuh-alerts-3.x-` but it also needs an index pattern to match these indices. The Wazuh App creates it once it is running for first time and saves it on `.kibana` index.  

Once you have a fine index pattern created, you need to generate at least one alert in order to create an indice with the prefix `wazuh-alerts-3.x-`, this is the main reason because we put a sample alert when installing.

**Give a try to this solution:**

_Delete only the right indices and restart the Kibana service. Notice that if you come from a major upgrade the indice .kibana could appear as .kibana-6. Also notice that deleting .kibana indice will erase all your custom visualizations, settings, etc._

```
# systemctl stop kibana
# curl -XDELETE elastic_ip:9200/.kibana
# curl -XDELETE elastic_ip:9200/.wazuh
# curl -XDELETE elastic_ip:9200/.wazuh-version
# systemctl start kibana
```

At this point you have deleted the main indices of Wazuh except your alerts, after it you have restarted Kibana so the Wazuh App is creating these indices again. Once it is done, restart the Wazuh manager, insert the sample alert or log out & log in, one of these actions will generate an alert, it's enough with one of them.

__**Recommended upgrade**__:

Upgrade to Wazuh 3.2 and the latest Wazuh App. Please check our [installation guide](https://documentation.wazuh.com/current/installation-guide/installing-elastic-stack/index.html) and check our App repository to read at [https://github.com/wazuh/wazuh-kibana-app/tree/3.2#upgrade](https://github.com/wazuh/wazuh-kibana-app/tree/3.2#upgrade) about which App version is compatible with your Elasticsearch installation. 

First step is to make it work with your current installation, after you have your environment with no troubles then make the upgrading process.

King regards,
Jesús

**Comment by JuanjiJG** [comment]
Hello @1522402210,

We would like to know if you're still facing this issue, or if you've already solved it. We'll be pleased to help you.

If you're using Wazuh 3, I suggest you try our latest available version, **Wazuh 3.2.1**, with several bugfixes and other improvements. You can read this [upgrading guide](https://documentation.wazuh.com/current/installation-guide/upgrading/latest_wazuh3_minor.html#upgrading-latest-minor) from our documentation to get more information.

Regards,
Juanjo

**Comment by JuanjiJG** [comment]
Hello @1522402210,

It's been more than a month since the last response to this thread. We'll proceed to close it, but if you need more assistance, don't hesitate to open it again or start a new issue, we'll be glad to help you.

Best regards,
Juanjo

**Comment by 1522402210** [comment]
@JuanjiJG  thanks
